control-plane/pkg/reconciler/consumergroup/consumergroup.go

```diff
@@ -421,6 +433,171 @@ func (r Reconciler) reconcileInitialOffset(ctx context.Context, cg *kafkainterna
 	return nil
 }
 
+func (r Reconciler) reconcileKedaObjects(ctx context.Context, cg *kafkainternals.ConsumerGroup) error {
+	// check if KEDA is installed
+	// TODO: code below failing unit tests with err: "panic: interface conversion: testing.ActionImpl is not testing.GetAction: missing method GetName"
+	/*if err := discovery.ServerSupportsVersion(r.KubeClient.Discovery(), keda.KedaSchemeGroupVersion); err != nil {
+		if strings.Contains(err.Error(), "server does not support API version") {
+			logging.FromContext(ctx).Errorw("KEDA not installed, failed to check API version", zap.Any("GroupVersion", keda.KedaSchemeGroupVersion))
+			return nil
+		}
+	}*/
+
+	_, err := r.KedaClient.KedaV1alpha1().ScaledObjects(cg.GetNamespace()).List(ctx, metav1.ListOptions{})
+	if err != nil && apierrors.IsNotFound(err) {
+		logging.FromContext(ctx).Errorw("KEDA not installed, failed to list ScaledObjects")
+		return nil
+	}
+
+	if !hasAuthSpecAuthConfig(cg.Spec.Template.Spec.Auth) && hasNetSpecAuthConfig(cg.Spec.Template.Spec.Auth) {
```

Comment:
this section is only relevant when auth is present/relevant, right? 

can we factor out to a function, instead of getting too long?